src/main/java/services/notification_sending/EmailNotificationPresenter.java

```diff
@@ -0,0 +1,81 @@
+package services.notification_sending;
+
+import java.util.Properties;
+import javax.mail.*;
+import javax.mail.internet.*;
+
+/**
+ * Sends out email notifications.
+ * Keeps a record of whether the user enabled the email notification settings.
```

Comment:
will every notif presenter be responsible for checking if its enabled or not? Would this not be a violation of DRY

Reply:
Yes, in the presentNotification method, enabled being true should lead to an early return.
Another way is to add and remove presenters in `observers` list of `NotificationTracker`.
We think record `enabled` in presenter itself might be easier to implement, since we don't need to create a new `presenter` object and add it to `observers` when enable it every time.